zanata-client-commands/src/main/java/org/zanata/client/commands/push/RawPushCommand.java

```diff
@@ -75,10 +94,164 @@ public RawPushCommand(PushOptions opts, RestClientFactory clientFactory) {
         client = getClientFactory().getFileResourceClient();
     }
 
+    /**
+     * Extract fileType from input pattern: fileType[extension;extension]
+     * @param typeWithExtension
+     */
+    private String extractFileType(String typeWithExtension) {
+        if (typeWithExtension.contains("[")) {
+            return typeWithExtension.substring(0,
+                    typeWithExtension.indexOf("[")).toUpperCase();
+        } else {
+            return typeWithExtension.toUpperCase();
+        }
+    }
+
+    /**
+     * Extract extensions in square bracket
+     * input pattern: fileType[extension;extension], [extension;extension]
+     */
+    private List<String> extractExtensions(String typeWithExtension) {
+        Pattern p = Pattern.compile("\\[(.*?)\\]");
```

Comment:
We could just use a single regular expression for file type and extensions. The advantage would be that the regex describes the whole pattern we are expecting so it is easier to understand what is expected from the code:

Something like this (first group is the type, second group is the extensions):

```
private static final Pattern fileNameExtensionsPattern = Pattern.complie("([^[*])(?:[(.*?)])");
```